File: BettingEngineServer/BettingEngineServer/Classes/EventOutcome.cs
using System.Collections.Generic;

namespace BettingEngineServer.Classes
{
    public class EventOutcome
    {

        public EventOutcome()
        {
            PossibleMarketOutcomes = new List<MarketOutcome>();
        }
        public string EventId { get; set; }
        public string WinningMarketId { get; set; }

        public Market WinningMarket { get; set; }
        //Negative amounts mean loss
        public decimal PLAmount { get; set; }
        public List<MarketOutcome> PossibleMarketOutcomes { get; set; }
    }
}

File: betting-engine-client/src/app/classes/event-outcome.ts
import {MarketOutcome} from './market-outcome';
import {Market} from './market';

export class EventOutcome {
  constructor() {
    this.possibleMarketOutcomes = [];
  }

  eventId: string;
  winningMarketId: string;
  winningMarket: Market;
  plAmount: number;
  possibleMarketOutcomes: MarketOutcome[];
}
